threesixty/tests/test_views.py

```diff
@@ -0,0 +1,278 @@
+import random
+import json
+from django.core import mail
+from django.urls import reverse
+
+from threesixty.views import AnswerCreateView
+from threesixty.forms import AnswerForm
+from threesixty.models import Survey, Participant, Question, Answer
+
+
+class TestViews:
+    def create_survey(self):
+        survey = Survey(
+            employee_name='sebastian',
+            employee_gender='male',
+            employee_email='sebastian@mail.com',
+            manager_email='johannes@mail.com',
+        )
+        survey.save()
+        return survey
+
+    def create_participant(self, survey_pk):
+        participant = Participant(
+            email='sebastian@mail.com',
+            survey_id=survey_pk,
+            relation='self',
+        )
+        participant.save()
+        return participant
+
+    def create_question(self):
+        question = Question(
+            text='how good is he?',
+            attribute='porfessionalitaet',
+            connotation=True,
+        )
+        question.save()
+        return question
+
+
+class TestSurveyDetailView(TestViews):
+    def test_get_survey(self, client, db):
+        survey = self.create_survey()
+        response = client.get(survey.get_absolute_url())
+
+        assert response.status_code == 200
+        assert response.context['object'].employee_name == 'sebastian'
+        assert response.context['object'].participant_can_skip is False
+
+    def test_get_survey_can_skip(self, db, client):
+        survey = self.create_survey()
+        survey.participant_can_skip = True
+        survey.save()
+
+        response = client.get(survey.get_absolute_url())
+
+        assert response.status_code == 200
+        assert response.context['object'].employee_name == 'sebastian'
+        assert response.context['object'].participant_can_skip is True
+
+
+class TestSurveyUpdateView(TestViews):
+    def test_get_update_view(self, client, db):
+        survey = self.create_survey()
+        survey.is_complete = True
+        survey.save()
+
+        response = client.get(survey.get_absolute_url())
+
+        assert response.status_code == 200
+        assert response.context['widget']['attrs']['checked']
+
+    def test_set_is_complete_true(self, client, db):
+        survey = self.create_survey()
+        survey.is_complete = False
+        survey.save()
+
+        response = client.post(survey.get_absolute_url(), {'is_complete': 'True'})
+
+        assert Survey.objects.get().is_complete
+
+    def test_set_is_complete_false(self, client, db):
+        survey = self.create_survey()
+        survey.is_complete = True
```

Comment:
This is why you shouldn't use persisted database fixtures. And you putting it in methods doesn't make it better.
I don't mind having a method that returns you an object, just don't call save on it. Do it in your test. Is way you are aware of the IO you are creating and that your tests will be slow.